forc-pkg/src/pkg.rs

```diff
@@ -779,6 +892,33 @@ pub fn graph_to_path_map(
     // Resolve all following dependencies, knowing their parents' paths will already be resolved.
     for dep_node in path_resolve_order {
         let dep = &graph[dep_node];
+        // If the dep is a path dependency check if it is removed.
+        if let SourcePinned::Path(_) = &dep.source {
+            // Retrieve the parent node to construct the relative path.
+            let (parent_node, _) = graph
+                .edges_directed(dep_node, Direction::Incoming)
+                .next()
+                .map(|edge| (edge.source(), edge.weight().clone()))
+                .ok_or_else(|| anyhow!("more than one root package detected in graph"))?;
+
+            let parent = &graph[parent_node].clone();
+            // Construct the path relative to the parent's path.
+            let parent_path = &path_map[&parent.id()];
+            let parent_manifest = ManifestFile::from_dir(parent_path, sway_git_tag)?;
+
+            // Check if parent's manifest file includes this dependency.
+            // If not this is a removed dependency and we should continue with the following dep
+            // Note that we need to check from package name as well since dependency name and package name is not the same thing
+            if parent_manifest.dep(&dep.name).is_none()
+                && !parent_manifest
+                    .deps()
+                    .filter_map(|dep| dep.1.package())
+                    .any(|package| package == dep.name)
+            {
+                removed_deps.insert(dep_node);
+                continue;
+            }
+        }
```

Comment:
This section repeats a lot of code that already exists in the `Path` branch within the following `match &dep_source` expression that immediately follows this block. Could we move this check to that branch, and re-use the existing retrieved `parent_manifest` etc?

Reply:
The problem that comes with moving this check to there is that, If this is indeed a removed dependency I don't want to continue to check its parent's manifest file (since we cannot find it) and just continue with the next dep. That's why I am checking this at the start of each iteration, if it is removed path dependency we continue with the next dependency.

I couldn't find a clean way of doing that while using that part of the code but open to any suggestion 🙌

Reply:
> If this is indeed a removed dependency I don't want to continue to check its parent's manifest file (since we cannot find it) and just continue with the next dep

I think we should still be able to do this in the following `match` expression with `continue`, just as you've done here? I don't believe there is anything else happening between this section and the `Path` branch below, so it should be fine?

Reply:
Yeah yeah, somehow I had the idea I couldn't do that but it was my mistake. This should be good to go now. Thanks  a lot 👍